docs/docs/events/builtin/extra/access_remote_vm.md

```diff
@@ -0,0 +1,34 @@
+# access_remote_vm
+
+## Intro
+access_remote_vm - gain access to the virtual memory of a separate process through the use of the procfs mem file.
+
+## Description
+This event marks access attempt of a process to the virtual memory of another process using the procfs mem file associated with that specific process (/proc/<pid>/mem).
```

Comment:
There is a signature in the open source called "proc_mem_code_injection" that checks for /proc/pid/mem file openings with WRITE flags (which @itaysk referred to some time ago). 

This signature is way more specific (picking a process doing reads from /proc/<pid>/mem and mapping to its own virtual address space).

While the 2 are different, with this one recognizing parts of the other process memory space that are being mapped, dealing with memory mapped addresses, etc, ... the first one (signature) would be triggered even before this event would be triggered, right ? 

So, in theory, this events are specific to a behavior that could be caught by that signature (probably with WAY less overhead than the signature). Could you confirm the understanding ?

Reply:
You are right - this is much easier way to implement these signatures.
In addition, it also allow to catch other flows in the kernel that access a process memory as a 3rd party - for example during execution, and with uprobes.
Above all else, this event give you much detailed information about the access, like the name of the memory mapped area accessed this way or the mmapped file.

Reply:
I believe the change should be in a separate PR and, if the other events can use a more performatic hook I'm all in (as checking for all opens is less performatic). Thanks for clarifying!